File: examples/dotNET/2. Functions/Geometric Matching/cs/ExampleImagesFolder.cs
using System;
using Microsoft.Win32;

namespace NationalInstruments.Vision
{
    /// <summary>
    /// Class to get the NI Vision example images folder
    /// </summary>
    public static class ExampleImagesFolder
    {
        public static string GetExampleImagesFolder()
        {
            RegistryKey key = Registry.LocalMachine;
            key = key.OpenSubKey(@"SOFTWARE\National Instruments\IMAQ Vision\CurrentVersion");
            string imagesPath = (string)key.GetValue("ExampleImagePath");
            key.Close();
            return imagesPath;
        }
    }
}

File: examples/dotNET/2. Functions/Color/Color Learn/vb/ExampleImagesFolder.vb
Imports System
Imports Microsoft.Win32

''' <summary>
''' Class to get the NI Vision example images folder
''' </summary>
Public NotInheritable Class ExampleImagesFolder
	Private Sub New()
	End Sub
	Public Shared Function GetExampleImagesFolder() As String
	    Dim key As RegistryKey = Registry.LocalMachine
	    key = key.OpenSubKey("SOFTWARE\National Instruments\IMAQ Vision\CurrentVersion")
	    Dim imagesPath As String = DirectCast(key.GetValue("ExampleImagePath"), String)
	    key.Close()
	    Return imagesPath
	End Function
End Class
